keps/sig-network/3698-multi-network-requirements/README.md

```diff
@@ -0,0 +1,375 @@
+#### Story #1
+As a Cloud Native Network Function (CNF) vendor I require an additional
+interface to be provisioned into the Kubernetes Pod. Each of these interfaces
+has to be in an isolated network for regulatory compliance. The isolation has to
+be done on a Layer-2.
+
+<p align="center">
+  <img src="mn-story-1.png?raw=true" alt="multi-network story 1 network L2 isolation"/>
+</p>
+
+#### Story #1a
+As a Cloud NativeNetwork Function (CNF) vendor I require an additional interface
+to be provisioned into the Kubernetes Pod. Each of these interfaces has to be in
+an isolated network for regulatory compliance. The isolation has to be done on a
+Layer-3.
+
+<p align="center">
+  <img src="mn-story-1a.png?raw=true" alt="multi-network story 1a network L3 isolation"/>
+</p>
+
+#### Story #2
+As a Cloud Native Network Function (CNF) vendor I require a HW-based interface
+(e.g. SRIOV VF) to be provisioned to my workload Pod. I need to leverage that HW
+for performance purposes (high bandwidth, low latency), that my user-space
+application (e.g. DPDK-based) can use. The VF will not use the standard netdev
+kernel module. The Pod’s scheduling to the nodes should be based on hardware
+availability (e.g. devicePlugin or some other way).
+
+<p align="center">
+  <img src="mn-story-2.png?raw=true" alt="multi-network story 2 HW interface"/>
+</p>
+
+#### Story #3
+I have implemented my Kubernetes cluster networking using a virtual switch. In
+this implementation I am capable of creating isolated Networks. I need a means
+to express to which Network my workloads connect to.
+
+<p align="center">
+  <img src="mn-story-3.png?raw=true" alt="multi-network story 3 virtual switch isolation"/>
+</p>
+
+#### Story #4
+As a Virtual Machine -based compute platform provider that I run on top of
+Kubernetes and Kubevirt I require multi-tenancy. The isolation has to be
+achieved on Layer-2 for security reasons.
+
+<p align="center">
+  <img src="mn-story-4.png?raw=true" alt="multi-network story 4 Kubevirt VMs"/>
+</p>
+
+#### Story #5
+As a platform operator I need to connect my on-premise networks to my workload
+Pods. I need to have the ability to represent these networks in my Kubernetes
+cluster in such a way that I can easily use them in my workloads.
+
+<p align="center">
+  <img src="mn-story-5.png?raw=true" alt="multi-network story 5 On-Premise Network representation"/>
+</p>
+
+#### Story #6
+As a Kubernetes cluster administrator I wish to isolate workloads based on
+namespaces and network access by overriding the cluster's default Network for a
+given Namespace. I do not want the tenants to change their manifests for that purpose.
+Those workloads should have the same level of Kubernetes functionality:
+Services, NetworkPolicies, access to Kubernetes API. I wish to support
+“profiles” for Namespace, where I can not only change default Network,
+but define a set of Networks assigned to given Namespace that Pods created in
+that NS are automatically attached to.
+
```

Comment:
OK... so there's a desire here to isolate workloads (traffic segregation).  This time by namespace.  It sounds like in this case there's a desire to have a segregated primary network by namespace, but its unclear to me from the requirement as written why this would imply a 'set of Networks'

It feels like this may be smashing two requirements together:

1.  A desire to have per-namespace traffic segmentation for primary networks.  If so this opens interesting questions about how that interacts with the rest of K8s networking
2. A desire to add a set of networks by namespace to a Pod... but its unclear *why* in this requirement.